packages/flutter/lib/src/animation/animation_controller.dart

```diff
@@ -68,7 +68,14 @@ enum AnimationBehavior {
   /// This is the default for repeating animations in order to prevent them from
   /// flashing rapidly on the screen if the widget does not take the
   /// [AccessibilityFeatures.disableAnimations] flag into account.
-  preserve,
+  preserve;
+
+  /// Whether animations should be enabled, based on the configured behavior
+  /// and the [AccessibilityFeatures.disableAnimations] flag.
+  bool get enableAnimations => switch (this) {
```

Comment:
This was designed as an enum so we could add additional behaviors later. `enableAnimations` pretends that it is just a boolean, which kinda defeats the purpose of that. If we add another value to AnimationBehavior we want the analyzer to tell us all the places we need to update with new behavior.